builtin/providers/aws/resource_aws_cloudwatch_metric_alarm.go

```diff
@@ -0,0 +1,266 @@
+		Update: resourceAwsCloudWatchMetricAlarmUpdate,
+		Delete: resourceAwsCloudWatchMetricAlarmDelete,
+
+		Schema: map[string]*schema.Schema{
+			"alarm_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"comparison_operator": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"evaluation_periods": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"metric_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"namespace": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"period": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"statistic": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"threshold": &schema.Schema{
+				Type:     schema.TypeFloat,
+				Required: true,
+			},
+			"actions_enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"alarm_actions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+			"alarm_description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"dimensions": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"insufficient_data_actions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+			"ok_actions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+			"unit": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudWatchMetricAlarmCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatchconn
+
+	params := getAwsCloudWatchPutMetricAlarmInput(d)
+
+	log.Printf("[DEBUG] CloudWatch PutMetricAlarm: %#v", params)
+	_, err := conn.PutMetricAlarm(&params)
+	if err != nil {
+		return fmt.Errorf("Error putting metric alarm: %s", err)
+	}
+	d.SetId(d.Get("alarm_name").(string))
+	log.Println("[INFO] CloudWatch Metric Alarm Created")
+
+	return resourceAwsCloudWatchMetricAlarmRead(d, meta)
+}
+
+func resourceAwsCloudWatchMetricAlarmRead(d *schema.ResourceData, meta interface{}) error {
+	a, err := getAwsCloudWatchMetricAlarm(d, meta)
+	if err != nil {
+		return err
+	}
+	if a == nil {
+		return nil
+	}
+
+	log.Printf("[DEBUG] Read Metric Alarm: %s", d.Get("alarm_name"))
+
+	d.Set("actions_enabled", a.ActionsEnabled)
+	d.Set("alarm_actions", a.AlarmActions)
```

Comment:
I believe that output you're sharing is from the plan or apply step? In which case, it's reading from the configuration, and using `d.Set` with a slice of strings (`[]string`). The case I'm referring to is when we read from AWS, and try using `d.Set` with what AWS returns, which is `[]*string`. 

For example, I took the configuration from your test example and plan/applied it, using `TF_LOG=1` to get the verbose output, and I get this:

    2015/05/18 14:14:12 terraform-provider-aws: 2015/05/18 14:14:12 [WARN] Error setting Alarm Actions: alarm_actions.0: '' expected type 'string', got unconvertible type '*string'

The same goes for `alarm_actions`. The action gets applied successfully, but if I modify an action in the console, refreshing the state locally shows this:

```
aws_cloudwatch_metric_alarm.bat: Refreshing state... (ID: terraform-test-foobar5)
2015/05/18 14:17:17 terraform-provider-aws: 2015/05/18 14:17:17 [DEBUG] Read Metric Alarm: terraform-test-foobar5
2015/05/18 14:17:17 terraform-provider-aws: 2015/05/18 14:17:17 [WARN] Error setting OK Actions: ok_actions.0: '' expected type 'string', got unconvertible type '*string'
```

As a result, the state file now looks like this:

```
  alarm_actions.# = 0
  alarm_description = This metric monitor ec2 cpu utilization
```